ArduinoCore-API/api/HardwareSPI.h

```diff
@@ -122,6 +123,8 @@ class HardwareSPI
     virtual void begin(const char *name);
 
     virtual void end();
+  private:
+    SPIChip *spiChip;
```

Comment:
I would either declare it protected or add a method for accessing spiChip by derived classes.